### PR TITLE
[WIP]cdc.redo: add DeleteAllLogs api in redolog storage

### PR DESCRIPTION
**Pull request by ben1009**
### What problem does this PR solve? 
add DeleteAllLogs api in redolog storage layer to support remove logs when remove changefeed

### Check List 

Tests 

 - Unit test

Code changes

 - Has interface methods change

Side effects

Related changes

 - Need to cherry-pick to the release branch
### Release note 

```release-note
None
```

### CODE REVIEW

File: cdc/redo/writer/blackhole_writer.go [hunk 1]
```diff
@@ -31,6 +31,11 @@ type blackHoleWriter struct {
 	checkpointTs uint64
 }
 
+// DeleteAllLogs ...
```
Comment (hi-rustin) [inline cdc/redo/writer/blackhole_writer.go:34]: Please refine the comment.

File: cdc/redo/writer/file.go [hunk 2]
```diff
@@ -403,56 +399,84 @@ func (w *Writer) rotate() error {
...
-	if !w.IsRunning() || w.isGCRunning() {
+	if w.isGCRunning() {
 		return nil
 	}
 
 	w.gcRunning.Store(true)
 	defer w.gcRunning.Store(false)
 
+	// if checkPointTs == math.MaxInt32 means remove all logs, stop write first
+	if checkPointTs == math.MaxInt32 {
```
Comment (amyangfei) [inline cdc/redo/writer/file.go:412]: Better to add a constant variable alias.

File: cdc/redo/writer/writer.go [hunk 3]
```diff
@@ -390,6 +396,50 @@ func (l *LogWriter) GetCurrentResolvedTs(ctx context.Context, tableIDs []int64)
 	return ret, nil
 }
 
+// 	DeleteAllLogs implement the DeleteAllLogs api
+func (l *LogWriter) DeleteAllLogs(ctx context.Context) error {
+	var err error
+	err = multierr.Append(err, l.rowWriter.GC(uint64(math.MaxUint32)))
+	err = multierr.Append(err, l.ddlWriter.GC(uint64(math.MaxUint32)))
```
Comment (amyangfei) [inline cdc/redo/writer/writer.go:403]: `math.MaxUint32` does not equal to the checked GC max ts.
Has the function `DeleteAllLogs` be tested?

File: cdc/redo/writer/file.go (hunk 2 above)
Comment (ben1009) [reply cdc/redo/writer/file.go]: done

File: cdc/redo/writer/writer.go (hunk 3 above)
Comment (ben1009) [reply cdc/redo/writer/writer.go]: tested but with api mocked, change to a const for maxTs

File: cdc/redo/writer/file.go [hunk 4]
```diff
@@ -403,72 +404,113 @@ func (w *Writer) rotate() error {
...
 	}
 
-	commitTs, fileType, err := common.ParseLogFileName(f.Name())
+	commitTs, fileType, err := common.ParseLogFileName(name)
 	if err != nil {
 		return false, err
 	}
 
-	return commitTs < checkPointTs && fileType == w.cfg.FileType, nil
+	return (tsShouldRM || commitTs < checkPointTs.Uint64()) && fileType == w.cfg.FileType, nil
```
Comment (hi-rustin) [inline cdc/redo/writer/file.go:502]: It might be better if we could explain it, or break it down into its corresponding variables to illustrate it!

**Review by hi-rustin — APPROVED** [review]
rest LGTM.

@amyangfei Do you think it is appropriate for us to use big int on checkpoint ts?

**Pull request closed without merging**